drivers/console/uart_console.c

```diff
@@ -108,12 +108,33 @@ static int console_out(int c)
 	return c;
 }
 
+/**
+ *
+ * @brief Get one character from UART
+ *
+ * @return The character received from console
+ */
+
+static int console_in(void)
+{
+	unsigned char c;
+
+	while (uart_poll_in(uart_console_dev, &c) < 0);
```

Comment:
This line looks intense, I guess we would probably want to sleep for a little between each poll, otherwise nothing runs? An interrupt-based implementation similar to the `shell_uart.c` would be nice

Reply:
Okay, I will try working on interrupt based implementation. 
Apart from this there were some minor bugs, I have taken care of them in the next patch

Reply:
@ycsin I have added the sleeping between poll. Will that be sufficient for now?
Thanks